consensus/reactor.go

```diff
@@ -948,14 +948,22 @@ func (ps *PeerState) ToJSON() ([]byte, error) {
 	return cdc.MarshalJSON(ps)
 }
 
-// GetHeight returns an atomic snapshot of the PeerRoundState's height
-// used by the mempool to ensure peers are caught up before broadcasting new txs
+// GetHeight returns an atomic snapshot of the PeerRoundState's height used by
+// the mempool to ensure peers are caught up before broadcasting new txs.
 func (ps *PeerState) GetHeight() int64 {
 	ps.mtx.Lock()
 	defer ps.mtx.Unlock()
 	return ps.PRS.Height
 }
 
+// GetLastBlockTime returns an atomic snapshot of the PeerRoundState's last
+// block time used by the evidence reactor when sending evidence.
+func (ps *PeerState) GetLastBlockTime() time.Time {
```

Comment:
Its not clear to me why they need the last block time? Why doesn't knowledge of the previous block hash suffice, if you already have the block?

Reply:
> if you already have the block?

what if we don't?

Reply:
But if they don't have the latest block, shouldn't I be tracking by block height which blocks to send to them?